### DISCUSSION TITLE
Generated Types with Fragments

### DISCUSSION THREAD

**Discussion by chaseconey** [dotansimha/graphql-code-generator#6096] [Q&A] [answered]
Hey there! Not sure if this is the right place to ask, happy to move this wherever it best fits. I am trying to find some assistance with type generation.

I have everything generating beautifully and am happy with the generation and configuration thus far. I seem to be running into one kind of odd typing issue having to do with fragment matching. I am making a query that splats on a fragment...something like this:

```gql
query DestinationQuery($path: String!) {
  route(path: $path) {
    ... on EntityCanonicalUrl {
      path
      entity {
        ... on NodePlace {
          title
          address
        }
      }
    }
  }
}
```

The types that are generated for this particular query seem to be doing an intersection on every possible type that can be generated in the route query rather than inferring the type based on the query. Is there a way to indicate which type to use for this specific query so that the generator can also infer it?

**Comment by dotansimha** [comment, marked as answer]
Hi @chaseconey ! Sorry for the late reply. 
You can find the answer for this output here: https://github.com/dotansimha/graphql-code-generator/issues/3950#issuecomment-864568310 (not the issue, just my comment intro :D )
